File: 13-02-2026/problem_8_graph_connectivity/correct/graph_connectivity.cpp
// PROBLEM 1: GRAPH CONNECTIVITY (BFS) - CORRECT C++
// ============================================
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n, m, q;
    cin >> n >> m >> q;
    
    vector<vector<int>> adj(n + 1);
    
    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    
    while (q--) {
        int s, t;
        cin >> s >> t;
        
        vector<int> vis(n + 1, 0);
        queue<int> qu;
        qu.push(s);
        vis[s] = 1;
        bool found = false;
        
        while (!qu.empty()) {
            int x = qu.front();
            qu.pop();
            
            if (x == t) {
                found = true;
                break;
            }
            
            for (int nb : adj[x]) {
                if (!vis[nb]) {
                    vis[nb] = 1;
                    qu.push(nb);
                }
            }
        }
        
        cout << (found ? "Yes\n" : "No\n");
    }
    
    return 0;
}


// =============================================

File: 13-02-2026/problem_8_graph_connectivity/correct/graph_connectivity.py
# PROBLEM 1: GRAPH CONNECTIVITY (BFS) - CORRECT PYTHON
# ============================================
import sys
from collections import deque

input = sys.stdin.readline

n, m, q = map(int, input().split())

adj = [[] for _ in range(n + 1)]

for _ in range(m):
    u, v = map(int, input().split())
    adj[u].append(v)
    adj[v].append(u)

for _ in range(q):
    s, t = map(int, input().split())
    
    vis = [False] * (n + 1)
    queue = deque([s])
    vis[s] = True
    found = False
    
    while queue:
        x = queue.popleft()
        
        if x == t:
            found = True
            break
        
        for nb in adj[x]:
            if not vis[nb]:
                vis[nb] = True
                queue.append(nb)
    
    print("Yes" if found else "No")


# =============================================